iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -607,41 +600,12 @@ await _internalRetryPolicy
             }
         }
 
-        public override Task OpenAsync(CancellationToken cancellationToken)
+        public override async Task OpenAsync(CancellationToken cancellationToken)
```

Comment:
What is the purpose of this method?

Reply:
To keep override implementation. As its used in all unit tests. 
https://github.com/Azure/azure-iot-sdk-csharp/blob/95532a519f6081e6987b04b7f8a571e7e1e98671/iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs#L65

Reply:
Can't the tests just be updated?

Reply:
Maybe what I really mean is, do we need both OpenAsync(CancellationToken) and OpenAsync(bool, CancellationToken). Can we combine them and just have one?

@timtay-microsoft and @abhipsaMisra, what does Java do? Can we have with retry be the default because the client has a retry policy?

Reply:
Java just has open(bool) since we agreed that it would be best for the user to be aware if their open call had retry baked in or not.

Reply:
If all the other methods have retry built-in, why not OpenAsync?

Reply:
I forget if there were any implementation related concerns in having retry baked into the Open call. If not, then I agree that it makes sense to have the Open call retry until it succeeds/the cancellation token expires.

Reply:
removed extra OpenSync() implementation and retry logic built-in as default.